### PR TITLE
Convert LockAdminUserWhenEditingUserTest to MFTF

### PR DESCRIPTION
**Pull request by Leandry**
### Description (*)

This PR with a test that checks lock admin user when editing admin user
Steps:

Steps:
1. Log in to backend as admin user.
2. Navigate to edit admin User Page.
3. Edit and save admin user data (current password is incorrect).
4. Perform action 6 specified number of times.
5. "The password entered for the current user is invalid. Verify the password and try again." appears after each attempt.
6. Check If the user was locked
### Fixed Issues 

#377 Convert LockAdminUserWhenEditingUserTest to MFTF 

### NOTE:
TODO:  Need to create delete operation for data entity="Admin3" in after section
To remove admin user after test execution

### CODE REVIEW

File: app/code/Magento/Backend/Test/Mftf/Section/AdminLoginFormSection.xml [hunk 1]
```diff
@@ -12,5 +12,6 @@
         <element name="username" type="input" selector="#username"/>
         <element name="password" type="input" selector="#login"/>
         <element name="signIn" type="button" selector=".actions .action-primary" timeout="30"/>
+        <element name="error" type="text" selector=".message.message-error.error"/>
```
Comment (rogyar) [inline app/code/Magento/Backend/Test/Mftf/Section/AdminLoginFormSection.xml]: Please, use the following element. It was recently merged into mainline:
https://github.com/magento/magento-functional-tests-migration/blob/6af0e0068cacd1d66a9f470c4cca5ae7454d0db4/app/code/Magento/Backend/Test/Mftf/Section/AdminLoginMessagesSection.xml#L12

File: app/code/Magento/User/Test/Mftf/ActionGroup/AdminEditUserActionGroup.xml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AdminEditUserActionGroup">
```
Comment (rogyar) [inline app/code/Magento/User/Test/Mftf/ActionGroup/AdminEditUserActionGroup.xml]: I would split this action group into more atomic components. For example:
- `AdminOpenUserEditPageActionGroup`
- `AdminUserEditFillFieldsActionGroup`
- `AdminUserEditAssignRoleActionGroup`
- `AdminUserEditSaveUserActionGroup`

File: dev/tests/functional/tests/app/Magento/Security/Test/TestCase/LockAdminUserWhenEditingUserTest.xml [hunk 3]
```diff
@@ -19,6 +19,7 @@
             <data name="user/data/password_confirmation" xsi:type="string">123123qq</data>
             <data name="user/data/current_password" xsi:type="string">incorrect password</data>
             <data name="attempts" xsi:type="string">4</data>
+            <data name="tag" xsi:type="string">mftf_migrated:yes</data>
```
Comment (rogyar) [inline dev/tests/functional/tests/app/Magento/Security/Test/TestCase/LockAdminUserWhenEditingUserTest.xml]: Please, add this tag as a comma-separated value to the existing tag on line 12.

File: app/code/Magento/Security/Test/Mftf/Test/LockAdminUserWhenEditingUserTest.xml [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+            <argument name="adminPasswordConfirmation" value="NEW$$user.password_confirmation$$"/>
+            <argument name="adminUserRole" value="1"/>
+            <argument name="messageSelector" value="{{AdminLoginFormSection.error}}" />
+            <argument name="message" value="Your account is temporarily disabled. Please try again later." />
+            <argument name="currentAdminPassword" value="{{_ENV.MAGENTO_ADMIN_PASSWORD}}INVALID"/>
+        </actionGroup>
+
+        <!-- Try to login as admin and check error -->
+        <actionGroup ref="LoginAsAdmin" stepKey="loginAsLockedAdmin"/>
+        <waitForPageLoad stepKey="waitForError"/>
```
Comment (rogyar) [inline app/code/Magento/Security/Test/Mftf/Test/LockAdminUserWhenEditingUserTest.xml]: Please, use the following action group for the similar checks

https://github.com/magento/magento-functional-tests-migration/blob/bc8efd2c46924888080186b30ecdcc435333074c/app/code/Magento/Backend/Test/Mftf/ActionGroup/AssertMessageOnAdminLoginActionGroup.xml#L11

**Comment by rogyar** [comment]
Hi @Leandry. Thank you for your collaboration. Please, check my comments in the review. Also, I would recommend merging the mainline (`2.3-develop`) into your current branch. There are many reusable components that might be useful within the scope of current test, a have mentioned these components in my comments as well. Thank you.

File: app/code/Magento/Backend/Test/Mftf/Section/AdminLoginFormSection.xml (hunk 1 above)
Comment (Leandry) [reply app/code/Magento/Backend/Test/Mftf/Section/AdminLoginFormSection.xml]: Done

File: app/code/Magento/User/Test/Mftf/ActionGroup/AdminEditUserActionGroup.xml (hunk 2 above)
Comment (Leandry) [reply app/code/Magento/User/Test/Mftf/ActionGroup/AdminEditUserActionGroup.xml]: Yes, Thank you. I've split action groups.

File: dev/tests/functional/tests/app/Magento/Security/Test/TestCase/LockAdminUserWhenEditingUserTest.xml (hunk 3 above)
Comment (Leandry) [reply dev/tests/functional/tests/app/Magento/Security/Test/TestCase/LockAdminUserWhenEditingUserTest.xml]: Done, thank you

**Comment by Leandry** [comment]
@rogyar, @okolesnyk  I added all requested changes. Could you check the changes?

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento-functional-tests-migration?pullRequest=704) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento-functional-tests-migration?pullRequest=704) before we can accept your contribution.<br/>**1** out of **9** committers have signed the CLA.<br/><br/>:white_check_mark: Leandry<br/>:x: danielrenaud<br/>:x: andimov<br/>:x: slopukhov<br/>:x: novikor<br/>:x: sivaschenko<br/>:x: alexander-aleman<br/>:x: danielgoodwin97<br/>:x: ysapiga

**Pull request merged**